### PR TITLE
Milestone3: Set Up Class Associations

### PR DESCRIPTION
**Pull request by Dinma-Faith-zz**
**This milestone was completed, following the requirements below:**
### Project requirements
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(date, person, book)
+    @rental = Rental.new(date, person, book)
+    @rental.student = person
+    @rental.book = book
+  end
```
Comment (gemmen29) [inline book.rb]: - [ ] Kindly update this method to be like
```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```
There is no need to pass the book as a parameter as you use the keyword `self` to access the book object.

File: person.rb [hunk 2]
```diff
@@ -27,4 +28,9 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals << rental unless @rentals.include?(rental)
+    rental.person = self
+  end
```
Comment (gemmen29) [inline person.rb]: - [ ] Kindly update this method to like:
```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```
as it's the same relation as between `Rental` and `Book`.

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
```
Comment (gemmen29) [inline classroom.rb:9]: - [ ] Kindly add the method that represents the `belongs to (a student belongs to a classroom)` inside `Student` class, you can find more info in this [article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Dinma-Faith, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi  @Dinma-Faith 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Great PR title
✔️ Good PR description
✔️ Commit messages are meaningful
✔️ No linter errors

# Status: Approved ✔️ 

simon

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Dinma-Faith-zz** [comment]
@gemmen29 Thanks for your corrections. I've been able to implement the changes. Thanks again

**Comment by Dinma-Faith-zz** [comment]
@KDlamini Thank you

**Pull request merged**